Utilities/Thread.cpp

```diff
@@ -1240,7 +1240,7 @@ bool handle_access_violation(u32 addr, bool is_writing, ucontext_t* context) noe
 
 	const auto cpu = get_current_cpu_thread();
 
-	if (rsx::g_access_violation_handler)
+	if (vm::check_addr(addr) && addr < RAW_SPU_BASE_ADDR && rsx::g_access_violation_handler)
```

Comment:
The range check should come first as it is a more trivial rejection. The likelihood that we would fault on a non-vm addr is very low.

```suggestion
	if (addr < RAW_SPU_BASE_ADDR && vm::check_addr(addr) && rsx::g_access_violation_handler)
```